### PR TITLE
Fix all lint warnings

### PR DESCRIPTION
**Pull request by nlarew**
```
> mongodb-rag-core

/drone/src/packages/mongodb-rag-core/src/PageFormat.ts
  1:10  warning  'logger' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

> mongodb-artifact-generator

/drone/src/packages/mongodb-artifact-generator/src/commands/generateDocsPage.ts
  61:7   warning  'embeddedContentStore' is defined but never used  @typescript-eslint/no-unused-vars
  61:29  warning  'embedder' is defined but never used              @typescript-eslint/no-unused-vars

/drone/src/packages/mongodb-artifact-generator/src/commands/parseDriversCsv.ts
   14:20  warning  'translateDocsPageAction' is defined but never used  @typescript-eslint/no-unused-vars
  146:46  warning  'pageAssets' is assigned a value but never used      @typescript-eslint/no-unused-vars
  185:37  warning  Unexpected any. Specify a different type             @typescript-eslint/no-explicit-any

> mongodb-rag-ingest

/drone/src/packages/mongodb-rag-ingest/src/embed/chunkOpenApiSpecYaml.ts
  49:55  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  98:55  warning  Forbidden non-null assertion              @typescript-eslint/no-non-null-assertion

/drone/src/packages/mongodb-rag-ingest/src/sources/LangchainDocumentLoaderDataSource.ts
  4:30  warning  'PageStore' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

/drone/src/packages/mongodb-rag-ingest/src/sources/mongodb-university/MongoDbUniversityDataApiClient.ts
   32:30  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  110:17  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  115:18  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/drone/src/packages/mongodb-rag-ingest/src/sources/mongodb-university/MongoDbUniversityDataSource.ts
  1:10  warning  'Page' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

/drone/src/packages/mongodb-rag-ingest/src/sources/snooty/SnootyDataSource.test.ts
  63:13  warning  'baseUrl' is assigned a value but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

> mongodb-chatbot-server

/drone/src/packages/mongodb-chatbot-server/src/routes/conversations/commentMessage.test.ts
  11:10  warning  'Db' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

/drone/src/packages/mongodb-chatbot-server/src/routes/conversations/getConversation.test.ts
  1:10  warning  'Db' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

/drone/src/packages/mongodb-chatbot-server/src/routes/conversations/rateMessage.test.ts
  11:14  warning  'MongoClient' is defined but never used. Allowed unused vars must match /^_/u       @typescript-eslint/no-unused-vars
  28:7   warning  'mongodb' is assigned a value but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

/drone/src/packages/mongodb-chatbot-server/src/services/mongodbConversations.ts
  1:10  warning  'SystemPrompt' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

/drone/src/packages/mongodb-chatbot-server/src/services/openAiChatLlm.ts
   3:3   warning  'ChatResponseMessage' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars
  12:3   warning  'OpenAiChatMessage' is defined but never used. Allowed unused vars must match /^_/u    @typescript-eslint/no-unused-vars
  15:10  warning  'Conversation' is defined but never used. Allowed unused vars must match /^_/u         @typescript-eslint/no-unused-vars

> chatbot-server-mongodb-public

/drone/src/packages/chatbot-server-mongodb-public/src/config.ts
  18:3   warning  'SystemPrompt' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars
  26:10  warning  'stripIndents' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

/drone/src/packages/chatbot-server-mongodb-public/src/index.ts
  39:28  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/drone/src/packages/chatbot-server-mongodb-public/src/middleware/blockGetRequests.test.ts
  4:61  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/drone/src/packages/chatbot-server-mongodb-public/src/processors/extractMetadataFromUserMessage.ts
  8:20  warning  'assert' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

/drone/src/packages/chatbot-server-mongodb-public/src/processors/makeStepBackRagGenerateUserPrompt.ts
  184:29  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/drone/src/packages/chatbot-server-mongodb-public/src/processors/makeStepBackUserQuery.ts
   57:29  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  115:29  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

> scripts

/drone/src/packages/scripts/src/createAug2023QualityTestsYaml.ts
  43:27  warning  'csvFilePath' is defined but never used. Allowed unused args must match /^_/u  @typescript-eslint/no-unused-vars

/drone/src/packages/scripts/src/main/upgradeFaqEntriesMain.ts
  5:22  warning  'fs' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

> mongodb-chatbot-evaluation

/drone/src/packages/mongodb-chatbot-evaluation/src/Pipeline.test.ts
   3:10  warning  'Pipeline' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars
  24:33  warning  Unexpected any. Specify a different type                                    @typescript-eslint/no-explicit-any

/drone/src/packages/mongodb-chatbot-evaluation/src/Pipeline.ts
  3:29  warning  'EvalConfig' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

/drone/src/packages/mongodb-chatbot-evaluation/src/evaluate/evaluateConversationAverageRetrievalScore.test.ts
  3:10  warning  'ConversationGeneratedData' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

/drone/src/packages/mongodb-chatbot-evaluation/src/evaluate/evaluateConversationFaithfulness.test.ts
   65:25  warning  Forbidden non-null assertion  @typescript-eslint/no-non-null-assertion
   65:25  warning  Forbidden non-null assertion  @typescript-eslint/no-non-null-assertion
  110:25  warning  Forbidden non-null assertion  @typescript-eslint/no-non-null-assertion
  110:25  warning  Forbidden non-null assertion  @typescript-eslint/no-non-null-assertion

/drone/src/packages/mongodb-chatbot-evaluation/src/evaluate/evaluateConversationRelevancy.test.ts
   65:25  warning  Forbidden non-null assertion  @typescript-eslint/no-non-null-assertion
   65:25  warning  Forbidden non-null assertion  @typescript-eslint/no-non-null-assertion
  109:25  warning  Forbidden non-null assertion  @typescript-eslint/no-non-null-assertion
  109:25  warning  Forbidden non-null assertion  @typescript-eslint/no-non-null-assertion

/drone/src/packages/mongodb-chatbot-evaluation/src/evaluate/evaluateExpectedLinks.test.ts
  127:34  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/drone/src/packages/mongodb-chatbot-evaluation/src/report/generateReportAndMetadata.test.ts
  9:8  warning  'e' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

/drone/src/packages/mongodb-chatbot-evaluation/src/test/mockConversations.ts
  2:3  warning  'AssistantMessage' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars
  7:3  warning  'SystemMessage' is defined but never used. Allowed unused vars must match /^_/u     @typescript-eslint/no-unused-vars
  8:3  warning  'UserMessage' is defined but never used. Allowed unused vars must match /^_/u       @typescript-eslint/no-unused-vars

/drone/src/packages/mongodb-chatbot-evaluation/src/test/mockReportFunc.ts
  14:3  warning  'reportName' is defined but never used. Allowed unused args must match /^_/u  @typescript-eslint/no-unused-vars

/drone/src/packages/mongodb-chatbot-evaluation/src/withConfig.test.ts
  6:35  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/drone/src/packages/mongodb-chatbot-evaluation/src/withConfig.ts
  115:6  warning  'Closeable' is defined but never used. Allowed unused vars must match /^_/u    @typescript-eslint/no-unused-vars
  119:6  warning  'CleanupFunc' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

```

### CODE REVIEW

File: packages/mongodb-rag-ingest/src/embed/chunkOpenApiSpecYaml.ts [hunk 1]
```diff
@@ -46,10 +46,19 @@ export const chunkOpenApiSpecYaml: ChunkFunc = async function (
   const { paths } = spec;
   if (paths !== undefined) {
     for (const path of Object.keys(paths)) {
-      const actions = paths[path] as { [key: string]: any };
+      // const actions = paths[path] as { [key: string]: any };
```
Comment (nlarew) [inline packages/mongodb-rag-ingest/src/embed/chunkOpenApiSpecYaml.ts:49]:
```suggestion
```

File: packages/mongodb-artifact-generator/src/commands/parseDriversCsv.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ import { promises as fs } from "fs";
 import path from "path";
 
 import { action as translateCodeAction } from "./translateCode";
-import { action as translateDocsPageAction } from "./translateDocsPage";
+// import { action as translateDocsPageAction } from "./translateDocsPage";
```
Comment (cbush) [inline packages/mongodb-artifact-generator/src/commands/parseDriversCsv.ts:14]: Still needed?

File: packages/mongodb-artifact-generator/src/commands/parseDriversCsv.ts [hunk 3]
```diff
@@ -143,7 +143,10 @@ export const action = createConfiguredAction<ParseDriversCsvCommandArgs>(
         }
       );
 
-      const { Code: codeExampleAssets, Page: pageAssets } = groupBy(
+      const {
+        Code: codeExampleAssets,
+        // Page: pageAssets
```
Comment (cbush) [inline packages/mongodb-artifact-generator/src/commands/parseDriversCsv.ts:148]: Needed?

**Review by cbush — APPROVED** [review]

File: packages/mongodb-artifact-generator/src/commands/parseDriversCsv.ts (hunk 2 above)
Comment (nlarew) [reply packages/mongodb-artifact-generator/src/commands/parseDriversCsv.ts:14]: Not really, removed

**Pull request merged**